### PR TITLE
fix collision between agent and bubble agent vehicle

### PR DESCRIPTION
**Pull request by JingfeiPeng**

### CODE REVIEW

File: smarts/core/vehicle_index.py [hunk 1]
```diff
@@ -382,7 +388,8 @@ def switch_control_to_agent(
                 is_hijacked=hijacking,
             )
         )
-
+        interface = sim.agent_manager.agent_interface_for_agent_id(agent_id)
+        print(f"Before: {agent_id} action: {interface}") # interfce none here
```
Comment (JingfeiPeng) [inline smarts/core/vehicle_index.py:392]: interface is None here

File: smarts/core/bubble_manager.py [hunk 2]
```diff
@@ -451,9 +451,11 @@ def _hijack_social_vehicle_with_social_agent(
             hijacking=True,
             recreate=False,
         )
-
+        interface = sim.agent_manager.agent_interface_for_agent_id(agent_id)
+        print(f"After: {agent_id} action: {interface.action}")
```
Comment (JingfeiPeng) [inline smarts/core/bubble_manager.py:455]: Interface is not none and can access the action

**Comment by Gamenot** [comment]
OK, I took a look at this and from what I can see it is a problem with using the TargetPose action space here.

**Comment by JingfeiPeng** [comment]
Smarts will only call controller's perform_action if the agent's action is in _dynamic_action_spaces.

We want the vehicle to have boxChassis so vehicle index.py, switch_control_to_agent check if agent's action is not in _dynamic_action_spaces and if so, create BoxChassis

To let the vehicle to reach the controller's perform_action method, we need to put TargetPose in dynnamic_action_spaces....

This causes vehicle index switch_control_to_agent to not create box model for the agents, and have the TargetPose agents to use AckermanChassis. Though this can be solved by making vehicle Index check if agent action is TargetPose or Multi-targetpose.

@Gamenot  But it seems the controllers are for vehicles with Ackerman chassis, and using MotionPlannerProvider for bubble agents with BoxChassis is that not bad of a choice?

**Comment by JingfeiPeng** [comment]
Finished ensuring collision issue is fixed and proper use of boxChassis for non dynamic agents.
Will create another issue for refactoring MotionPlannerProvider into a controller similar to Trajectory_Tracking controller

File: smarts/core/vehicle.py [hunk 3]
```diff
@@ -483,6 +498,14 @@ def sync_to_panda3d(self):
         pos, heading = self._chassis.pose.as_panda3d()
         self._np.setPosHpr(*pos, heading, 0, 0)
 
+    @lru_cache(maxsize=1)
+    def _warn_non_kinematic_vehicle_set_pose(self):
```
Comment (Gamenot) [inline smarts/core/vehicle.py:502]: This was a bad naming scheme on my part. Also not called.

File: smarts/core/vehicle.py [hunk 4]
```diff
@@ -483,6 +498,14 @@ def sync_to_panda3d(self):
...
+        if self._has_stepped and isinstance(self._chassis, AckermannChassis):
+            logging.warning(
+                f"Agent `{self._id}` has called set pose after step."
+                "This may cause collision problems"
+            )
+
     # TODO: Merge this w/ speed setter as a set GCD call?
     def set_pose(self, pose: Pose):
```
Comment (Gamenot) [inline smarts/core/vehicle.py:510]: I think here ^.

**Review by Gamenot — APPROVED** [review]

File: smarts/core/chassis.py [hunk 5]
```diff
@@ -164,9 +165,7 @@ def dimensions(self) -> BoundingBox:
 
     @property
     def contact_points(self) -> Sequence:
-        contact_points = _query_bullet_contact_points(
-            self._client, self._bullet_body, -1
-        )
+        contact_points = _query_bullet_contact_points(self._client, self.bullet_id, -1)
```
Comment (Gamenot) [inline smarts/core/chassis.py:168]:
```suggestion
        contact_points = _query_bullet_contact_points(self._client, self._bullet_body.bullet_id, -1)
```

**Review by Gamenot — CHANGES_REQUESTED** [review]

**Review by Gamenot — APPROVED** [review]

**Review by iman512003 — APPROVED** [review]

**Pull request merged**